bears/general/AnnotationBear.py

```diff
@@ -28,104 +27,212 @@ def run(self, filename, file, language: str, coalang_dir: str = None):
                                 respectively.
         """
         lang_dict = LanguageDefinition(language, coalang_dir=coalang_dir)
-        # Strings
-        # TODO treat single-line and multiline strings differently
-        strings = dict(lang_dict["string_delimiters"])
-        strings.update(lang_dict["multiline_string_delimiters"])
-        strings_found = self.find_with_start_end(filename, file, strings)
-        # multiline Comments
-        comments_found = self.find_with_start_end(
-                filename, file, dict(lang_dict["multiline_comment_delimiters"]))
-        # single-line Comments
-        comments_found.update(self.find_singleline_comments(
-                          filename, file, list(lang_dict["comment_delimiter"])))
-
-        matches_found = strings_found | comments_found
-        # Remove Nested
-        unnested_annotations = set(filter(
-                                       lambda arg: not starts_within_ranges(
-                                           arg, matches_found),
-                                       matches_found))
-        # Yield different results for strings and comments
-        strings_found = tuple(filter(lambda arg: arg in unnested_annotations,
-                                     strings_found))
-        comments_found = tuple(filter(lambda arg: arg in unnested_annotations,
-                                      comments_found))
-        yield HiddenResult(self, {'comments': comments_found,
-                                  'strings': strings_found})
+        string_delimiters = dict(lang_dict["string_delimiters"])
+        multiline_string_delimiters = dict(
+            lang_dict["multiline_string_delimiters"])
+        multiline_comment_delimiters = dict(
+            lang_dict["multiline_comment_delimiters"])
+        comment_delimiter = dict(lang_dict["comment_delimiter"])
+        strings = comments = ()
+        try:
+            strings, comments = self.find_annotation_ranges(
+                file,
+                filename,
+                string_delimiters,
+                multiline_string_delimiters,
+                comment_delimiter,
+                multiline_comment_delimiters)
 
-    @staticmethod
-    def find_with_start_end(filename, file, annot):
+        except NoCloseError as e:
+            yield Result(self, str(e), severity=RESULT_SEVERITY.MAJOR,
+                         affected_code=(e.code,))
+
+        content = {"strings": strings, "comments": comments}
+        yield HiddenResult(self, content)
+
+    def find_annotation_ranges(self, file, filename,
+                               string_delimiters, multiline_string_delimiters,
+                               comment_delimiter, multiline_comment_delimiters):
         """
-        Gives all positions of annotations which have a start and end.
+        Find ranges of all annotations.
+
+        :param file:
+            A tuple of strings, with each string being a line in the file.
```

Comment:
i saw it with the `JSHintBear` and used it with` RuboCopBear` and i found this way to be more cleaner, readable? since now i can write in some sort of a paragraph, also this allows for examples, though there are none right now :P, i'll cjhange it back if necessary though